api/context.go

```diff
@@ -523,3 +526,15 @@ func InvalidateAllCaches() {
 	store.ClearChannelCaches()
 	store.ClearUserCaches()
 }
+
+func (c *Context) checkTeamId() {
+	if c.TeamId != ""  && c.Session.GetTeamByTeamId(c.TeamId) == nil {
+		if HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
+			if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
+				c.Err = result.Err
+				c.Err.StatusCode = http.StatusBadRequest
+				return
+			}
+		}
```

Comment:
I think you need an else condition for `if HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {` that also sets an error to handle the non-system admin case. Something like:

```
if c.TeamId != ""  && c.Session.GetTeamByTeamId(c.TeamId) == nil {
	if HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
		if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
			c.Err = result.Err
 			c.Err.StatusCode = http.StatusBadRequest
 			return
 		}
 	}
        c.Err = model.NewLocAppError(<add parameters here>)
        c.Err.StatusCode = http.StatusBadRequest
 	return
}
```